pkg/controller/pxc/users.go

```diff
@@ -0,0 +1,143 @@
+package pxc
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"os"
+
+	api "github.com/percona/percona-xtradb-cluster-operator/pkg/apis/pxc/v1"
+	"github.com/percona/percona-xtradb-cluster-operator/pkg/pxc/users"
+	"github.com/pkg/errors"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (r *ReconcilePerconaXtraDBCluster) reconcileUsers(cr *api.PerconaXtraDBCluster) error {
+	if cr.Status.Status != api.AppStateReady {
+		return nil
+	}
+
+	if len(cr.Spec.Users.Secrets) == 0 {
+		return nil
+	}
+	secretObj := corev1.Secret{}
+	err := r.client.Get(context.TODO(),
+		types.NamespacedName{
+			Namespace: cr.Namespace,
+			Name:      cr.Spec.SecretsName,
+		},
+		&secretObj,
+	)
+	if err != nil && k8serrors.IsNotFound(err) {
+		return nil
+	} else if err != nil {
+		return errors.Wrapf(err, "get cluster secret '%s'", cr.Spec.SecretsName)
+	}
+
+	for _, secretName := range cr.Spec.Users.Secrets {
+		operator := corev1.Pod{}
```

Comment:
can we fetch it only if when we run user-manager (not on every reconcile loop)

Reply:
done https://github.com/percona/percona-xtradb-cluster-operator/pull/342/commits/74d8e58743302032ea789cd46dfb590f31384fef